src/components/QuickGrid/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { QuickGrid } from '../QuickGrid';
+
+import * as _ from 'lodash';
+import { GridColumn } from '../QuickGrid.Props';
+import { Checkbox } from '../../Checkbox';
+
+import './ColumnPicker.scss';
+import { Icon } from '../../Icon';
+import { Button } from '../../Button';
+
+export interface ColumnPickerProps {
+    columns: Array<GridColumn>;
+    pickedColumns?: Array<GridColumn>;
+
+    onDismiss?(event?: React.MouseEvent<HTMLElement>);
+    onChanged?(picked: Array<GridColumn>);
+}
+
+export interface ColumnPickerState {
+    isColumnSelected: {[key: string]: boolean};
+}
+
+
+export default class ColumnPicker extends React.Component<ColumnPickerProps, ColumnPickerState> {
+
+    public constructor(props: ColumnPickerProps) {
+        super(props);
+        this.state = {
+            isColumnSelected: this.setVisibleColumns(props.columns, props.pickedColumns)
+        };
+    }
+
+    private setVisibleColumns(columns: Array<GridColumn>, picked?: Array<GridColumn>): {[key: string]: boolean} {
+        let visible: {[key: string]: boolean} = {};
+        columns.forEach(col => visible[col.valueMember] = !!!picked);
```

Comment:
why the triple !!! instead of a single?

Reply:
It's an optional field, so it's first converted to boolean `!!(undefined)` and then inverted

Reply:
It's working with a single !. This is not the first time I used this operator :joy: